Packet++/header/SomeIpSdLayer.h

```diff
@@ -0,0 +1,771 @@
+#ifndef PACKETPP_SOMEIPSD_LAYER
+#define PACKETPP_SOMEIPSD_LAYER
+
+#include "EndianPortable.h"
+#include "IpAddress.h"
+#include "Layer.h"
+#include "SomeIpLayer.h"
+#include <cstring>
+#include <iterator>
+#include <map>
+#include <memory>
+#include <stdexcept>
+#include <vector>
+
+/// @file
+
+/**
+ * \namespace pcpp
+ * \brief The main namespace for the PcapPlusPlus lib
+ */
+namespace pcpp
+{
+/**
+ * Types of protocols that can be referenced in SOME/IP-SD
+ */
+enum SomeIpSdProtocolType : uint8_t
+{
+	/** TCP */
+	SD_TCP = 0x06,
+	/** UDP */
+	SD_UDP = 0x11
+};
+
+class SomeIpSdLayer;
+
+/**
+ * @class SomeIpSdOption
+ * Base class of the SOME/IP-SD options. Cannot be instantiated.
+ */
+class SomeIpSdOption
+{
+public:
+	friend class SomeIpSdLayer;
+
+	/**
+	 * Types of options currently available for the SOME/IP-SD protocol
+	 */
+	enum class OptionType : uint8_t

```

Comment:
i added the 'Unknown' option type, but i think this is not needed currently

Reply:
It might be used here: https://github.com/kolbex/PcapPlusPlus/blob/ac36a0c9bc953b04ad8ba72f378503a8572cd491/Packet%2B%2B/src/SomeIpSdLayer.cpp#L22

If `getSomeIpSdOptionHeader()->type` has an invalid value